Core/Systems/ModPlayers/UniversalBuffingPlayer.cs

```diff
@@ -17,5 +20,28 @@ public override void PostUpdateEquips()
 	public override void ResetEffects()
 	{
 		UniversalModifier = new EntityModifier();
+		OnFireChance = StatModifier.Default;
+	}
+	
+	/// <summary>
+	/// Used to apply on hit effects for affixes that have them
+	/// </summary>
+	/// <param name="target"></param>
+	/// <param name="hit"></param>
+	/// <param name="damageDone"></param>
+	public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
+	{
+		if (OnFireChance.Base > 0)
+		{
+			float chance = Main.rand.NextFloat();
+			if (chance > OnFireChance.Base)
+			{
+				return;
```

Comment:
Well, we might/will most likely have more on-hit things so return is probably not right; but we can also just fix this in the future.